rep_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 The go-zeromq Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package zmq4_test
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/go-zeromq/zmq4"
+)
+
+func TestIssue99(t *testing.T) {
+	var (
+		wg     sync.WaitGroup
+		outMsg zmq4.Msg
+		inMsg  zmq4.Msg
+	)
+
+	ep, err := EndPoint("tcp")
+	if err != nil {
+		t.Errorf("could not find endpoint: %v", err)
+	}
+
+	requester := func() {
+
+		req := zmq4.NewReq(context.Background())
+		defer req.Close()
+		defer wg.Done()
+
+		err := req.Dial(ep)
+		if err != nil {
+			t.Errorf("could not dial: %v", err)
+		}
+
+		// Test message w/ 3 frames
+		outMsg = zmq4.NewMsgFromString([]string{"ZERO", "Hello!", "World!"})
+		err = req.Send(outMsg)
+		if err != nil {
+			t.Errorf("failed to send: %v", err)
+		}
+
+		inMsg, err = req.Recv()
+		if err != nil {
+			t.Errorf("failed to recv: %v", err)
```

Comment:
```suggestion
			t.Errorf("failed to recv: %+v", err)
			return
```

Reply:
Come on...

Reply:
well, the `%+v` can be quite instrumental in giving the whole "stack" of errors (via the `errors.Unwrap` mechanism).
and the `return` is needed to replicate the original `t.Fatal` code flow.